### DISCUSSION TITLE
Project 1 Question

### DISCUSSION THREAD

**Discussion by ecieszynski** [cooperrc/computational-mechanics#31] [Q&A] [answered]
Here is my code for 1 and 2 for the first project. I feel as though I am missing some aspect of the question, because it felt almost too easy. Any guidance would be appreciated

![image](https://user-images.githubusercontent.com/106542680/172065029-f0de3c14-b81e-45c6-ba67-3f437e08b0b8.png)

**Comment by caroline-cook** [comment]
I did a very similar process as you so I hope that we both have the right idea!

**Comment by cooperrc** [comment, marked as answer]
> Here is my code for 1 and 2 for the first project. I feel as though I am missing some aspect of the question, because it felt almost too easy. Any guidance would be appreciated

If its feels easy _and_ you have an answer, that's great!

To clarify a little (we'll get more into depth for numerical derivatives in Mod_03+05):

You are _approximating_ a derivative with a [_finite difference_ method](https://cooperrc.github.io/computational-mechanics/module_05/01_Revisiting_derivatives.html#revisiting-derivatives). There are two equally valid ways to do this with 2 data points:

1. backward difference method: $\frac{dT}{dt} \approx \frac{T_1 - T_{0}}{\Delta t} = -k(T_1 - T_A)$ e.g. current time is 1pm
2. forward difference method: $\frac{dT}{dt} \approx \frac{T_{1} - T_{0}}{\Delta t} = -k(T_0 - T_A)$ e.g. current time is 11am

In our case, this means there are **2** equally valid approximations for $k$. You can try calculating both  if you want (there's not a _huge_ difference).
